src/lang-go.ts

```diff
@@ -189,31 +190,22 @@ async function tryToCreateAccessToken(): Promise<string | undefined> {
 }
 
 async function connectAndInitialize(
-    address: string,
+    address: URL,
     root: URL,
     token: string | undefined
 ): Promise<rpc.MessageConnection> {
-    const connectingToGoLangserverHelp = [
-        `Unable to connect to the Go language server at ${address}.`,
-        `Make sure ${'go.address' as keyof Settings} in your Sourcegraph settings is set to the address of the language server (e.g. wss://sourcegraph.example.com/go).`,
-        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-go`,
-    ].join('\n')
-
-    const connectingToSourcegraphHelp = [
-        `The Go language server running on ${address} was unable to fetch repository contents from Sourcegraph running on ${sourcegraphURL()}.`,
-        `Make sure ${'go.sourcegraphUrl' as keyof Settings} in your settings is set to the address of Sourcegraph from the perspective of the language server (e.g. http://sourcegraph-frontend:30080 when running in Kubernetes).`,
-        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-go`,
-    ].join('\n')
-
-    const connection = (await new Promise((resolve, reject) => {
+    const connection = (await new Promise<wsrpc.MessageConnection>((resolve, reject) => {
         try {
-            const webSocket = new WebSocket(address)
+            const webSocket = new WebSocket(address.href)
             const conn = createWebSocketConnection(wsrpc.toSocket(webSocket), new ConsoleLogger())
             webSocket.addEventListener('open', () => resolve(conn))
-            webSocket.addEventListener('error', event => reject(new Error(connectingToGoLangserverHelp)))
+            webSocket.addEventListener('error', event => {
+                notifyUnableToConnectToLanguageServer(address)
```

Comment:
This seems redundant - this rejects the Promise, which is caught in the `catch` Block, where this function is also called. In general it's not good to have this logic in the event handler because if `notifyUnableToConnectToLanguageServer` throws itself for some reason that error will not be propagated and the Promise will be pending forever.